### PR TITLE
Buffs the Hot Potato Bomb

### PR DESCRIPTION
**Pull request by redmoogle**
# Document the changes in your pull request

Instead of detonating after 60 seconds it detonates after 10 seconds of not being forced onto someone else
Increases Explosiveness (Will blow the user up into a pile of parts)
Changed TC cost to 12
And some code cleaning
image

# Wiki Documentation

Tc Cost Increase (4 -> 12)
Balance Change

# Changelog
:cl:  
tweak: Hot potato now costs 12 TC
tweak: Hot potato now explodes after 10 seconds if not forced onto someone
tweak: Hot potato is now deadly explosive
/:cl:

### CODE REVIEW

**Comment by Moggram** [comment]
The fact that there is a 60 sec timer is part of the fun of using it.

**Comment by redmoogle** [comment]
> The fact that there is a 60 sec timer is part of the fun of using it.

Makes it useless as an item

**Comment by Moggram** [comment]
> Makes it useless as an item

[Its not always about killing your target](https://youtu.be/Du1Xxmxd5Ww), [its a meme redmoogle](https://youtu.be/T6r4-LsVKG4)

File: code/game/objects/items/hot_potato.dm [hunk 1]
```diff
@@ -151,14 +153,15 @@
...
-	detonation_timerid = null
 	colorize(null)
 	active = FALSE
 
 /obj/item/hot_potato/update_icon()
-	icon_state = active? icon_on : icon_off
+	if(active)
+		icon_state = "[initial(icon_state)]_active"
+	else
+		icon_state = initial(icon_state)
```
Comment (tattax) [inline code/game/objects/items/hot_potato.dm]:
```suggestion
	icon_state = "[initial(icon_state)]" + "[active ? "active" : ""]"
```

File: code/game/objects/items/hot_potato.dm [hunk 2]
```diff
@@ -151,14 +153,12 @@
...
 	REMOVE_TRAIT(src, TRAIT_NODROP, HOT_POTATO_TRAIT)
-	deltimer(detonation_timerid)
 	STOP_PROCESSING(SSfastprocess, src)
-	detonation_timerid = null
 	colorize(null)
 	active = FALSE
 
 /obj/item/hot_potato/update_icon()
-	icon_state = active? icon_on : icon_off
+	icon_state = "[initial(icon_state)]" + "[active ? "active" : ""]"
```
Comment (tattax) [inline code/game/objects/items/hot_potato.dm]:
```suggestion
	icon_state = "[initial(icon_state)]" + "[active ? "_active" : ""]"
```
forgot this

**Comment by Rdrazga** [comment]
> > The fact that there is a 60 sec timer is part of the fun of using it.
> 
> Makes it useless as an item

I mean the whole meme of it was supposed to be its a hot potato, and was extraordinarily cheap. True buff would be making it play hot potato music as long as its active and you are near the carrier.

**Comment by redmoogle** [comment]
> > > The fact that there is a 60 sec timer is part of the fun of using it.
> > 
> > 
> > Makes it useless as an item
> 
> I mean the whole meme of it was supposed to be its a hot potato, and was extraordinarily cheap. True buff would be making it play hot potato music as long as its active and you are near the carrier.

thats probably an amazing way to get hit with a copyright claim

**Comment by Moggram** [comment]
> thats probably an amazing way to get hit with a copyright claim

Mahahahaa and how did that work out for Fox trying to kill CM ss13?

**Comment by redmoogle** [comment]
> 
> Mahahahaa and how did that work out for Fox trying to kill CM ss13?

Unnecessary legal risk.

**Comment by Rdrazga** [comment]
> > Mahahahaa and how did that work out for Fox trying to kill CM ss13?
> 
> Unnecessary legal risk.

Ok it doesn't need to be THE hot potato song but same concept.

**Comment by SynnGraffkin** [comment]
I feel like 12 TC is a lot for a single explosive you can stick on someone. 
The cost could probably be adjusted to 8, which matches a bag of X-4, and perhaps increase the timer to 10 or 15 seconds. 
12 TC for a 5 second bomb, if no one else is around, that can gib is kinda 'spensive.

**Comment by redmoogle** [comment]
> I feel like 12 TC is a lot for a single explosive you can stick on someone. The cost could probably be adjusted to 8, which matches a bag of X-4, and perhaps increase the timer to 10 or 15 seconds. 12 TC for a 5 second bomb, if no one else is around, that can gib is kinda 'spensive.

theres a decent difference between x4 and this. namely that attacking someone with it will force drop items and make it latch on to them until they click some one else. As for the timer I feel like 5s is decent but maybe 7-9s might be better. I might change the tc cost to 10

**Comment by JamieD1** [comment]
maybe make it slightly longer then 5

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**